src/v2/Components/SaleDetailTimer.tsx

```diff
@@ -0,0 +1,110 @@
+import { SaleDetailTimer_sale } from "v2/__generated__/SaleDetailTimer_sale.graphql"
+import { createFragmentContainer, graphql } from "react-relay"
+import * as React from "react"
+import { Flex, Text } from "@artsy/palette"
+import { useTimer } from "v2/Utils/Hooks/useTimer"
+import moment from "moment"
+export interface SaleDetailTimerProps {
+  sale: SaleDetailTimer_sale
+}
+
+function relativeDaysUntil(parsedDays, suffix) {
+  if (parsedDays === 0) {
+    return `${suffix} Today`
+  } else if (parsedDays === 1) {
+    return `${parsedDays} Day Until ${suffix}`
+  } else {
+    return `${parsedDays} Days Until ${suffix}`
+  }
+}
+
+export const SaleDetailTimer: React.FC<SaleDetailTimerProps> = ({ sale }) => {
+  const endAt = sale?.endAt
+
+  const startAt = sale?.startAt
+
+  const endedAt = sale?.endedAt
+
+  const { time } = useTimer(endAt!, startAt!)
+
+  if (!endAt) {
+    return null
+  }
+
+  const timerCopy = getTimerCopy(time, startAt, endAt, endedAt)
+
+  return (
+    <Flex alignItems="center" flexDirection="column">
+      <Text variant="lg" color={"blue100"}>
+        <Text color={timerCopy.color}>{timerCopy.copy}</Text>
+      </Text>
+    </Flex>
+  )
+}
+
+export const SaleDetailTimerFragmentContainer = createFragmentContainer(
+  SaleDetailTimer,
+  {
+    sale: graphql`
+      fragment SaleDetailTimer_sale on Sale {
+        endAt
+        endedAt
+        startAt
+      }
+    `,
+  }
+)
+
+export const getTimerCopy = (time, startAt, endAt, endedAt) => {
+  const { days, hours, minutes, seconds } = time
+
+  const parsedDays = parseInt(days, 10)
+  const parsedHours = parseInt(hours, 10)
+  const parsedMinutes = parseInt(minutes, 10)
+  const parsedSeconds = parseInt(seconds, 10)
+  const thisMoment = moment()
```

Comment:
Did you try using the `useTimer` hook for all of this logic? I don't think you need to use the `moment` library at all here. `useTimer` can provide a value for `hasStarted` that you can use instead. The hook handles the "current time" concept for you.

Reply:
I tried that early on and struggled with it, but I see what I did wrong and will be able to refactor without using those. Good call!